tests/nested/manual/core20-remodel/task.yaml

```diff
@@ -0,0 +1,116 @@
+summary: verify a simple UC20 remodel
+
+systems: [ubuntu-20.04-64]
+
+environment:
+  NESTED_CUSTOM_MODEL: $TESTSLIB/assertions/valid-for-testing-pc-20.model
+  NESTED_IMAGE_ID: uc20-remodel-testing
+  NESTED_ENABLE_TPM: true
+  NESTED_ENABLE_SECURE_BOOT: true
+  NESTED_BUILD_SNAPD_FROM_CURRENT: true
+
+prepare: |
+    #shellcheck source=tests/lib/nested.sh
+    . "$TESTSLIB/nested.sh"
+
+    # modify and repack gadget snap to add a defaults section and use our own
+    # prepare-device hook to use the fakedevicesvc
+    mkdir "$(nested_get_extra_snaps_path)"
+
+    # Get the snakeoil key and cert for signing gadget assets (shim)
+    KEY_NAME=$(nested_get_snakeoil_key)
+    SNAKEOIL_KEY="$PWD/$KEY_NAME.key"
+    SNAKEOIL_CERT="$PWD/$KEY_NAME.pem"
+
+    snap download --basename=pc --channel="20/edge" pc
+    unsquashfs -d pc-gadget pc.snap
+
+    nested_secboot_sign_gadget pc-gadget "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"
+    snap pack pc-gadget/ "$(nested_get_extra_snaps_path)"
+
+    tests.nested build-image core
+    tests.nested create-vm core
+
+execute: |
+    # shellcheck source=tests/lib/nested.sh
+    . "$TESTSLIB/nested.sh"
+    boot_id="$( nested_get_boot_id )"
+    tests.nested exec snap model |MATCH 'model +my-model$'
+    # XXX: recovery system label is based on a date; we may end up with a
+    # different label if the remodel runs around midnight; the label will
+    # conflict with an existing system label
+    label_base=$(tests.nested exec "date '+%Y%m%d'")
+
+    echo "Refresh model assertion to revision 2"
+    nested_copy "$TESTSLIB/assertions/valid-for-testing-pc-revno-2-20.model"
+    REMOTE_CHG_ID="$(tests.nested exec sudo snap remodel --no-wait valid-for-testing-pc-revno-2-20.model)"
+    nested_wait_for_reboot "${boot_id}"
+    tests.nested exec sudo snap watch "${REMOTE_CHG_ID}"
+
+    echo "Verify the system is back in run mode"
+    tests.nested exec "sudo cat /proc/cmdline" | MATCH snapd_recovery_mode=run
+
+    # this is a simple model revision refresh
+    echo "Verify the model"
+    tests.nested exec snap model --verbose > rev-2-model
+    MATCH 'model: +my-model' < rev-2-model
+    MATCH 'revision: +2' < rev-2-model
+
+    # seed system was created
```

Comment:
can we also test that `snap recovery` works here too?

Reply:
Added